### PR TITLE
fix(rule ticket): issue with SLA criteria

### PR DESCRIPTION
**Pull request by Rom1-B**
With a rule of this type that has an SLA criterion. If only the SLA is updated on the ticket, the rule applies, but if only the urgency is updated, the rule does not apply.

image

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !27501

### CODE REVIEW

File: src/Ticket.php [hunk 1]
```diff
@@ -1250,11 +1250,15 @@ public function prepareInputForUpdate($input)
             list($dateField, $slaField) = SLA::getFieldNames($slmType);
             if (isset($input[$slaField]) && ($input[$slaField] > 0)) {
                 $manual_slas_id[$slmType] = $input[$slaField];
+            } elseif (!isset($input[$slaField])) {
+                $input[$slaField] = $this->fields[$slaField];
```
Comment (cedric-anne) [inline src/Ticket.php:1254]: IMHO, to help people to understand why these values are forced, added code should be moved in `fillInputForBusinessRules` and a comment should be added.

**Review by cedric-anne — COMMENTED** [review]
@orthagh 

Do we want to always push SLA/OLA in the rules input?

**Comment by orthagh** [comment]
seems ok to do an exception for these fields.
But for recall, rules, for ticket update only manage currently (apart few exceptions) user inputs and not the current state of a ticket.
And so why not urgency, impact, etc ?

Maybe we could think about a more global solution for next major as this "bug" come back frequently.
Like adding a checkbox in rule "Only check modified fields" checked by default to match the current behavior, uncheking it filling all data ?

**Review by cedric-anne — COMMENTED** [review]
Seems OK. Could you add a test for this bug?

File: src/Ticket.php [hunk 2]
```diff
@@ -6443,6 +6443,18 @@ private function fillInputForBusinessRules(array &$input)
         if (count($calendars)) {
             $input['_date_creation_calendars_id'] = $calendars;
         }
+
+        // add SLA/OLA (for business rules)
+        foreach ([SLM::TTR, SLM::TTO] as $slmType) {
```
Comment (cedric-anne) [inline src/Ticket.php:6448]: It should probably be surrounded by `if (!$this->isNewItem()) { ... }`

**Review by cedric-anne — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Pull request merged**